### DISCUSSION TITLE
Input following a regular expression

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#12587] [Feature Requests]
A NTH feature: Be able to set a regular expression for the input field.
Use cases: 
- a date input with only the month and the year: `[01][0-9]\/[12][0-9][0-9][0-9]`
- SSN `(?!666|000|9\\d{2})\\d{3}-(?!00)\\d{2}-(?!0{4})\\d{4}`
- CRN `(SC|NI|[0-9]{2})[0-9]{6}`
- ...

**Comment by rijkvanzanten** [comment]
Would the regex be for value validation, or input masking?

**Comment by Pachat** [comment]
input masking is better

**Comment by Pachat** [comment]
Input masking could also avoid a misuse of `Input` as `Integer`
If you set an `Input` as `Integer` in Directus typing non integers are still allowed.

For instance, if you type `123 k$`, the Save button is not activated (correct as you typed non integers), but the user sees `123 k$` he typed.
Suppose that after changing some other field he is now enabled to click the Save button. 
What happens is that the `123 k$` he still sees will not be saved.

With input masking applied to `Input` as `Integer`, he would simply be allowed to type only what is going to be saved.

**Comment by cristianocardoso0** [comment]
This is an essential feature for any basic application.

Include possibility to create masks in input fields and save the field as raw or masked.

Example:


+55 47 99999-9999
99.999.999/9999-99

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
